detect_regressions.py

```diff
@@ -0,0 +1,115 @@
+
+    # regression analysis only on tests that passed
+    df = pd.read_sql(
+        "select * from test_run where platform = 'linux' and call_outcome = 'passed'",
+        engine,
+    )
+
+    # join runtime + py version
+    df = df.assign(
+        runtime=(
+            "coiled-"
+            + df.coiled_runtime_version
+            + "-py"
+            + df.python_version.str.split(".", n=2).str[:2].str.join(".")
+        ),
+        category=df.path.str.split("/", n=1).str[0],
+    )
+
+    regressions = []
+    reg_df = pd.DataFrame(
+        columns=["category", "type", "mean", "last", "last-1", "last-2", "threshold"]
+    )
+
+    runtimes = list(df.runtime.unique())
+    for runtime in runtimes:
+        by_test = df[(df.runtime == runtime)].groupby("name")
+
+        test_names = list(by_test.groups.keys())
+        for name in test_names:
+            df_test = by_test.get_group(name)
+
+            # check the test is not obsolete.
+            if pd.Timestamp(df_test.start.iloc[-1]) < pd.Timestamp.now() - pd.Timedelta(
+                days=15
+            ):
+                # the latest run was 7+ days ago, test is obsolete
+                pass
+            else:
+                for metric in ["duration", "average_memory", "peak_memory"]:
+                    # check that we have enough data to do some stats
+                    # currently we don't have a lot of data but eventually use 10 instead of 6.
+                    if len(df_test[metric]):
+                        category = df_test.category.unique()[0]
+                        metric_threshold = (
+                            df_test[metric][-13:-3].mean()
+                            + 2 * df_test[metric][-13:-3].std()
+                        )
+
+                        if (df_test[metric].iloc[-3:] >= metric_threshold).all():
+
+                            if metric in ["average_memory", "peak_memory"]:
+                                units_norm = 1 / (
+                                    1024**3
+                                )  # to GiB to match dashboard
+                                u = "[GiB]"
+                            else:
+                                units_norm = 1
+                                u = "[s]"
+
+                            last_three = (
+                                df_test[metric].iloc[-1] * units_norm,
+                                df_test[metric].iloc[-2] * units_norm,
+                                df_test[metric].iloc[-3] * units_norm,
+                            )
+                            reg = (
+                                f"{runtime = }, {name = }, {category = }, "
+                                f"last_three_{metric} {u} = "
+                                f"{last_three}, "
+                                f"{metric}_threshold {u} = {metric_threshold * units_norm} \n"
+                            )
+
+                            regressions.append(reg)
```

Comment:
These summary strings could also be a column in `reg_df`, might make it slightly easier to pass the dataframe around as we incorporate this into the static site.